File: frontend/src/types/screenshot.ts
export type ProcessingStatus =
  | "pending"
  | "processing"
  | "completed"
  | "failed";

export interface ScreenshotSummary {
  id: string;
  filename: string;
  status: ProcessingStatus;
}

export interface UploadScreenshotItem {
  id: string;
  filename: string;
  status: ProcessingStatus;
}

export interface UploadResponse {
  uploaded: number;
  screenshots: UploadScreenshotItem[];
}

export interface ScreenshotListResponse {
  screenshots: ScreenshotSummary[];
}

export interface QueryRequest {
  query: string;
}

export interface QuerySource {
  id: string;
  filename: string;
  relevance: number;
}

export interface QueryResponse {
  answer: string;
  sources: QuerySource[];
  found: boolean;
}

File: backend/app/schemas/screenshot.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field

from app.models.screenshot import ProcessingStatus


class UploadScreenshotItem(BaseModel):
    id: str
    filename: str
    status: ProcessingStatus


class UploadResponse(BaseModel):
    uploaded: int
    screenshots: List[UploadScreenshotItem]


class ScreenshotSummary(BaseModel):
    id: str
    filename: str
    status: ProcessingStatus


class ScreenshotListResponse(BaseModel):
    screenshots: List[ScreenshotSummary]


class ScreenshotDetail(BaseModel):
    id: str
    filename: str
    file_path: str
    status: ProcessingStatus
    ocr_text: Optional[str] = None
    image_description: Optional[str] = None
    processing_error: Optional[str] = None
    created_at: datetime
    updated_at: datetime
